internals/overlord/overlord.go

```diff
@@ -64,21 +64,42 @@ type Overlord struct {
 	pruneTicker *time.Ticker
 
 	// managers
-	inited     bool
-	runner     *state.TaskRunner
-	serviceMgr *servstate.ServiceManager
-	commandMgr *cmdstate.CommandManager
-	checkMgr   *checkstate.CheckManager
-	logMgr     *logstate.LogManager
+	inited           bool
+	runner           *state.TaskRunner
+	serviceMgr       *servstate.ServiceManager
+	commandMgr       *cmdstate.CommandManager
+	checkMgr         *checkstate.CheckManager
+	logMgr           *logstate.LogManager
+	externalManagers map[any]StateManager
 }
 
-// New creates a new Overlord with all its state managers.
+// ManagerProvider is the interface that ManagerGenerator depends on
+//
+// Overlord implements ManagerProvider as it provides the necessary
+// handles to hook an external manager into the overlord's environment.
+type ManagerProvider interface {
```

Comment:
I went with 1. but then realized I was creating boilerplate code implementing those functions when I could just be using the ones of Overlord directly.

The reason we're not going with 3. is to avoid exposing the whole Overlord, something which Gustavo insisted on after seeing the previous version.